particula/next/gas/vapor_pressure.py

```diff
@@ -0,0 +1,370 @@
+
+"""Strategy pattern for vapor pressure calculations.
+All units are in base SI units (kg, m, s).
+"""
+
+from abc import ABC, abstractmethod
+from typing import Union
+from numpy.typing import NDArray
+import numpy as np
+
+from particula.constants import GAS_CONSTANT  # pyright: ignore
+
+
+def calculate_partial_pressure(
+            concentration: Union[float, NDArray[np.float_]],
+            molar_mass: Union[float, NDArray[np.float_]],
+            temperature: Union[float, NDArray[np.float_]]
+        ) -> Union[float, NDArray[np.float_]]:
+    """
+    Calculate the partial pressure of a gas from its concentration, molar mass,
+    and temperature.
+
+    Parameters:
+    - concentration (float): Concentration of the gas in kg/m^3.
+    - molar_mass (float): Molar mass of the gas in kg/mol.
+    - temperature (float): Temperature in Kelvin.
+
+    Returns:
+    - float: Partial pressure of the gas in Pascals (Pa).
+    """
+    # Calculate the partial pressure
+    return (
+        concentration * float(GAS_CONSTANT.m) * temperature) / molar_mass
+
+
+def calculate_concentration(
+    partial_pressure: Union[float, NDArray[np.float_]],
+    molar_mass: Union[float, NDArray[np.float_]],
+    temperature: Union[float, NDArray[np.float_]]
+) -> Union[float, NDArray[np.float_]]:
+    """
+    Calculate the concentration of a gas from its partial pressure, molar mass,
+    and temperature using the ideal gas law.
+
+    Parameters:
+    - pressure (float or NDArray[np.float_]): Partial pressure of the gas
+    in Pascals (Pa).
+    - molar_mass (float or NDArray[np.float_]): Molar mass of the gas in kg/mol
+    - temperature (float or NDArray[np.float_]): Temperature in Kelvin.
+
+    Returns:
+    - concentration (float or NDArray[np.float_]): Concentration of the gas
+    in kg/m^3.
+    """
+    return (partial_pressure * molar_mass
+            ) / (float(GAS_CONSTANT.m) * temperature)
+
+
+class VaporPressureStrategy(ABC):
```

Comment:
I would like to keep, the Abstract class in the same file, as the concrete implementations. That seems easier to read/understand.